gc/base/MemorySubSpaceGenerational.hpp

```diff
@@ -107,6 +107,8 @@ class MM_MemorySubSpaceGenerational : public MM_MemorySubSpace
 
 	virtual uintptr_t releaseFreeMemoryPages(MM_EnvironmentBase* env);
```

Comment:
Not needed for Generational - we must pass memory type always since we have to make a decision what child subspaced we will pass the request too.
This is opposite from what is needed for its  child subspaces (Flat/SemiSpace/Generic), that should keep this variant of the API, but don't need the variant with parameters. They make not decisions and just pass the request to its own child or there is not child pass to memory pool, for all types that they deal with (in fact, they are not even aware of memory types).